packages/admin/src/form/FinalFormSelect.tsx

```diff
@@ -1,14 +1,46 @@
 import { SelectProps } from "@material-ui/core";
+import { MenuItem } from "@material-ui/core";
 import * as React from "react";
 import { FieldRenderProps } from "react-final-form";
 
 import { Select } from "./Select";
 
-interface IProps extends FieldRenderProps<string, HTMLInputElement | HTMLTextAreaElement> {}
+interface FinalFormSelectProps<T> extends FieldRenderProps<T, HTMLInputElement | HTMLTextAreaElement> {
+    optionLabelKey?: keyof T;
+}
 
-// SelectProps also has a input prop, which collides with final-form input
-export const FinalFormSelect: React.FunctionComponent<IProps & Omit<SelectProps, "input">> = ({
+export const FinalFormSelect = <T extends Record<string, any>>({
     input: { checked, value, name, onChange, onFocus, onBlur, ...restInput },
     meta,
+    optionLabelKey = "label",
+    options = null,
+    isAsync = false,
+    loading = false,
     ...rest
-}) => <Select {...rest} name={name} onChange={onChange} value={value} onFocus={onFocus} onBlur={onBlur} />;
+}: FinalFormSelectProps<T> & Omit<SelectProps, "input">) => {
+    if (options === null) {
+        return <Select {...rest} name={name} onChange={onChange} value={value} onFocus={onFocus} onBlur={onBlur} />;
+    }
+
+    const getOptionLabel = (option: T) => {
+        if (!optionLabelKey || option[optionLabelKey] === undefined) {
+            throw new Error(`"${optionLabelKey}" is not a key in the options-prop, please set the prop "optionLabelKey" accordingly.`);
```

Comment:
throwing an error is a bit hard... probably console.error and the fallback you had before would also do it?

Reply:
Now it behaves like in mui-autocomplete: no value is shown and an console.error is shown.